[build.webkit.org] Convert CompileWebKit to new style buildstep

aj062 opened the pull request:
#### a5574a1d36d5b02c339ae191e812eda03a89e42a
<pre>
[build.webkit.org] Convert CompileWebKit to new style buildstep
<a href="https://bugs.webkit.org/show_bug.cgi?id=296290">https://bugs.webkit.org/show_bug.cgi?id=296290</a>
<a href="https://rdar.apple.com/156340170">rdar://156340170</a>

Reviewed by Brianna Fan.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit):
(CompileWebKit.start): Converted to run()
(CompileWebKit.buildCommandKwargs): Deprecated in newer buildbot version. Achieved same functionality in __init__.
(CompileWebKit._addToLog): Deleted.
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_custom_timeout_specified_in_factory): Added new unit-test for timeout specified in factory.

Canonical link: <a href="https://commits.webkit.org/298343@main">https://commits.webkit.org/298343@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f6b5533edfd7ebb13b85768e0126ea78ecee2fd1

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115035 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34751 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25232 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121155 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65808 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/9e68687f-b466-43b7-83ff-5b4faf9d5986) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116924 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35396 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43312 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87417 "Found 14 new test failures: imported/w3c/web-platform-tests/css/css-text-decor/text-shadow/basic-opacity.html imported/w3c/web-platform-tests/css/css-text-decor/text-shadow/basic.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-auto-last-word-001.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-character.html imported/w3c/web-platform-tests/css/css-text/hyphens/hyphens-vs-float-clearance-002.html imported/w3c/web-platform-tests/css/css-transforms/animation/transform-non-invertible-discrete-interpolation.html imported/w3c/web-platform-tests/css/css-transforms/transform-box/cssbox-content-box-002.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-left-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/massive-element-on-top-of-viewport-offscreen-new.html imported/w3c/web-platform-tests/css/css-view-transitions/pseudo-with-classes-exit.html ... (failure)") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/e6823fb4-bb2f-45c1-a8c7-5159446a5fe3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117983 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28210 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103277 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67813 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27383 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21397 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64807 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97592 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21514 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124345 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42008 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31412 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96215 "Found 1 new test failure: imported/w3c/web-platform-tests/service-workers/service-worker/redirected-response.https.html (failure)") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/114468 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42380 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99467 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96000 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41202 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19039 "Passed tests") | [❌ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/38016 "Hash f6b5533e for PR 48338 does not build (failure)") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18434 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41883 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47418 "Build is in progress. Recent messages:") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41427 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44744 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43163 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

briannafan commented on Tools/CISupport/build-webkit-org/steps.py:
```diff
@@ -409,11 +410,13 @@ def start(self):
 
-        return shell.Compile.start(self)
+        rc = yield super().run()
+        rc = self.evaluateCommandCustom(rc)
```
Why do we need to define this as `evaluateCommandCustom` and call it explicitly instead of just writing `evaluateCommand`? IIRC this is what we usually do.

aj062 replied:
Right. I double-checked and `evaluateCommand` works normally in new style steps as well. Updated the PR to retain evaluateCommand().

briannafan reviewed (approved):
Should we be using `_add_to_log` instead of `print` in line 437? Just happened to notice this while reviewing. Otherwise, LGTM

aj062 committed:
[build.webkit.org] Convert CompileWebKit to new style buildstep
https://bugs.webkit.org/show_bug.cgi?id=296290
rdar://156340170

Reviewed by Brianna Fan.

* Tools/CISupport/build-webkit-org/steps.py:
(CompileWebKit):
(CompileWebKit.start): Converted to run()
(CompileWebKit.buildCommandKwargs): Deprecated in newer buildbot version. Achieved same functionality in __init__.
(CompileWebKit._addToLog): Deleted.
* Tools/CISupport/build-webkit-org/steps_unittest.py:
(TestCompileWebKit.test_custom_timeout_specified_in_factory): Added new unit-test for timeout specified in factory.

Canonical link: https://commits.webkit.org/298343@main
